### PR TITLE
ovl_Bg_F40_Flift decompiled

### PR DESCRIPTION
**Pull request by sonicdcer**

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_F40_Flift/z_bg_f40_flift.c [hunk 1]
```diff
@@ -28,28 +31,67 @@ const ActorInit Bg_F40_Flift_InitVars = {
...
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Flift/BgF40Flift_Destroy.s")
+void func_808D7714(BgF40Flift* this, PlayState* play) {
+    if (this->unk_160 != 0) {
+        this->unk_160--;
+        this->dyna.actor.world.pos.y =
+            (((cos_rad(this->unk_160 * 0.032724924f) * this->dyna.actor.params) + 1.0f) * 463.4f) +
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Bg_F40_Flift/z_bg_f40_flift.c:79]: `(M_PI / 96.0f)`

File: src/overlays/actors/ovl_Bg_F40_Flift/z_bg_f40_flift.c [hunk 2]
```diff
@@ -28,28 +31,67 @@ const ActorInit Bg_F40_Flift_InitVars = {
...
+    } else {
+        if (this->unk_160 == 0) {
+            this->unk_160 = 0x30;
+        }
+        this->unk_160--;
+        if (this->dyna.actor.params == 1) {
+            this->dyna.actor.world.pos.y = (sin_rad(this->unk_160 * 0.1308997f) * 5.0f) + this->dyna.actor.home.pos.y;
+        } else {
+            this->dyna.actor.world.pos.y =
+                (sin_rad(this->unk_160 * 0.1308997f) * 5.0f) + (926.8f + this->dyna.actor.home.pos.y);
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Bg_F40_Flift/z_bg_f40_flift.c:70]: `(M_PI / 24.0f)`

File: src/overlays/actors/ovl_Bg_F40_Flift/z_bg_f40_flift.c [hunk 3]
```diff
@@ -28,28 +31,67 @@ const ActorInit Bg_F40_Flift_InitVars = {
...
+void BgF40Flift_Update(Actor* thisx, PlayState* play) {
+    BgF40Flift* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Flift/func_808D7714.s")
+    this->actionFunc(this, play);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Flift/BgF40Flift_Update.s")
+void BgF40Flift_Draw(Actor* thisx, PlayState* play) {
+    BgF40Flift* this = THIS;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Bg_F40_Flift/z_bg_f40_flift.c:94]: Shouldn't be required.

File: src/overlays/actors/ovl_Bg_F40_Flift/z_bg_f40_flift.c [hunk 4]
```diff
@@ -28,28 +31,67 @@ const ActorInit Bg_F40_Flift_InitVars = {
...
+void func_808D75F0(BgF40Flift* this, PlayState* play) {
+    if (((this->dyna.actor.params == 1) && (DynaPolyActor_IsInRidingMovingState(&this->dyna))) ||
+        ((this->dyna.actor.params == -1) && (!DynaPolyActor_IsInRidingMovingState(&this->dyna)))) {
+        this->unk_160 = 0x60;
+        this->actionFunc = func_808D7714;
+    } else {
+        if (this->unk_160 == 0) {
+            this->unk_160 = 0x30;
+        }
+        this->unk_160--;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Bg_F40_Flift/z_bg_f40_flift.c:65]: Looks like this is a timer.

**Review by hensldm — APPROVED** [review]
Though needs a format.

File: src/overlays/actors/ovl_Bg_F40_Flift/z_bg_f40_flift.c [hunk 5]
```diff
@@ -28,28 +31,65 @@ const ActorInit Bg_F40_Flift_InitVars = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Flift/BgF40Flift_Destroy.s")
+void func_808D75F0(BgF40Flift* this, PlayState* play) {
+    if (((this->dyna.actor.params == 1) && (DynaPolyActor_IsInRidingMovingState(&this->dyna))) ||
+        ((this->dyna.actor.params == -1) && (!DynaPolyActor_IsInRidingMovingState(&this->dyna)))) {
+        this->timer = 0x60;
+        this->actionFunc = func_808D7714;
+    } else {
+        if (this->timer == 0) {
+            this->timer = 0x30;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Bg_F40_Flift/z_bg_f40_flift.c:63]: Timer, so dec.

File: src/overlays/actors/ovl_Bg_F40_Flift/z_bg_f40_flift.c [hunk 6]
```diff
@@ -28,28 +31,65 @@ const ActorInit Bg_F40_Flift_InitVars = {
...
+        }
+        this->timer--;
+        if (this->dyna.actor.params == 1) {
+            this->dyna.actor.world.pos.y = (sin_rad(this->timer * 0.1308997f) * 5.0f) + this->dyna.actor.home.pos.y;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Bg_F40_Flift/z_bg_f40_flift.c:67]: This constant appeared elsewhere in the file.

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**